libpam/include/pam_inline.h

```diff
@@ -175,4 +175,23 @@ pam_read_passwords(int fd, int npass, char **passwords)
 	return i;
 }
 
+static inline int
+pam_consttime_streq(const char *userinput, const char *secret) {
+	volatile const char *u = userinput, *s = secret;
+	volatile int ret = 0;
+
+	while (*u != '\0') {
+		ret |= *u++ ^ *s;
+
+		if (*s != '\0')
+			s++;
+		else
+			s = secret;
```

Comment:
Maybe also avoid branching:
```suggestion
		s += !!*s;
```